packages/lit-virtual/src/index.ts

```diff
@@ -0,0 +1,108 @@
+import { ReactiveController, ReactiveControllerHost } from 'lit'
+import {
+  PartialKeys,
+  Virtualizer,
+  VirtualizerOptions,
+  observeWindowOffset,
+  observeWindowRect,
+} from '@tanstack/virtual-core'
+import {
+  observeElementOffset,
+  observeElementRect,
+  windowScroll,
+  elementScroll,
+} from '@tanstack/virtual-core'
+
+class VirtualizerControllerBase<
+  TScrollElement extends Element | Window,
+  TItemElement extends Element,
+> implements ReactiveController
+{
+  host: ReactiveControllerHost
+
+  private readonly virtualizer: Virtualizer<TScrollElement, TItemElement>
+
+  private cleanup: () => void = () => {}
+
+  constructor(
+    host: ReactiveControllerHost,
+    options: VirtualizerOptions<TScrollElement, TItemElement>,
+  ) {
+    ;(this.host = host).addController(this)
+
+    const resolvedOptions: VirtualizerOptions<TScrollElement, TItemElement> = {
+      ...options,
+      onChange: (instance, sync) => {
+        instance._willUpdate()
+        this.host.updateComplete.then(() => this.host.requestUpdate())
+        options.onChange?.(instance, sync)
+      },
+    }
+    this.virtualizer = new Virtualizer(resolvedOptions)
+  }
+
+  public getVirtualizer() {
+    return this.virtualizer
+  }
+
+  async hostConnected() {
+    this.cleanup = this.virtualizer._didMount()
+    this.virtualizer._willUpdate()
+    await this.host.updateComplete
+    this.host.requestUpdate()
+    this.virtualizer.measure()
```

Comment:
Actually I started with this approach at the beginning, but for some reason in the Columns example, if failed to size the columns right, it would fix only after a scroll event was triggered.

Reply:
Hey @piecyk, it's been a while :) do we plan to move forward with this?

Reply:
@kadoshms please checkout https://github.com/TanStack/virtual/pull/737 wit this change we can simplify the adapter  

https://github.com/TanStack/virtual/assets/82964/5cfee50a-1233-424b-b8d1-54bcc6493618

Reply:
Awesome 👍 
I'll check this out merged with my branch and report back this week

Reply:
@piecyk for column & grid dynamic virutalizers that works great - however I noticed a small glitch with row - if you scroll down to the very bottom, there's some weird behavior:
![scroll](https://github.com/TanStack/virtual/assets/5352933/184596c7-ab4c-4f76-a296-228beaa34cb5)

Reply:
It's a bit hard to see it with the gif

Reply:
> It's a bit hard to see it with the gif

hmm can't really reproduce it, on what browser and os you are getting this? Btw changed a bit the impalemntion on https://github.com/TanStack/virtual/pull/737 to make it simpler added the `measureElement` back on virtual item.

Reply:
@piecyk unfortunately now without the explicit `measure`, seems like the column sizes are not calculated correctly, and only "fixed" when a scroll event is triggered:
![Uploading image.png…]()

Why is using "measure" explicitly, only once when the element is connected to the DOM is so problematic? 
The solid adapter takes the same approach from what Iv'e seen.

Reply:
@kadoshms hmm running virtual-core with this changes against lit and works fine for me.

> Why is using "measure" explicitly, only once when the element is connected to the DOM is so problematic?
The solid adapter takes the same approach from what Iv'e seen.

the issue is that the virtualizer measureElement is based on node.isConnected, so when lit is calling ref with node it will be false so we don't measure and don't register RO callback. 

So for example if you scroll down as side effect it looks like it's working, but for example if jump on the end of the list and start scrolling up it will jump as we can adjust the scroll when element size changes.

Reply:
@piecyk did you try it with your proposed changes on the column virtualizer?

Reply:
Yes, for initial rows, when user is scrolling elements are added/removed calling ref with node.isConnected false that will not attach RO etc, basic will lead to unexpected behaviour in dynamic case.

Reply:
Yes, so it works nicely now - but I still need to keep `measure` on mount, do we want to progress with the current implementation then?

Reply:
@piecyk I see - I'll try to check how dynamic cases work in this case (probably later on this week though 😄 )

Reply:
@kadoshms merge the `measureElement` on virtual item, you can rebase with main and update the lit adapter, let's merge it soon.

Reply:
Works great now :)